src/seccompiler/src/compiler.rs

```diff
@@ -58,6 +60,43 @@ impl fmt::Display for Error {
     }
 }
 
+/// Deserializable object that represents the Json filter file.
+pub(crate) struct JsonFile(pub HashMap<String, Filter>);
+
+// Implement a custom deserializer, that returns an error for duplicate thread keys.
+impl<'de> Deserialize<'de> for JsonFile {
+    fn deserialize<D>(deserializer: D) -> result::Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        struct MyVisitor;
+
+        impl<'d> Visitor<'d> for MyVisitor {
```

Comment:
Should we rename MyVisitor with something else like VisitorWrapper or i don't know

Reply:
Renamed it to `JsonFileVisitor`